cores/esp32/HardwareSerial.cpp

```diff
@@ -128,7 +128,8 @@ HardwareSerial::HardwareSerial(int uart_nr) :
 _uart_nr(uart_nr), 
 _uart(NULL), 
 _rxBufferSize(256), 
-_onReceiveCB(NULL), 
+_onReceiveCB(NULL),
+_onReceiveTimeout(10),
```

Comment:
This is the only thing I disagree. By default now (as before my pull request), onReceive callback is being called by timeout (10 symbos by default) or by rx fifo full (120 bytes). That is the standard behaviour by IDF. I don't think that Arduino core should use another default. I would prefer to let the timeout on because if you receive 119 bytes, you will never get called (unless you receive on more byte that will trigger UART_INTR_RXFIFO_FULL interrupt)